docs/content/develop/set-up-dev-environment.md

```diff
@@ -0,0 +1,82 @@
+---
+title: "Set up your development environment"
+weight: 12
+---
+
+# Set up your development environment
+
+Before you start adding or updating a Terraform resource using
+[magic-modules]({{< ref "/" >}}), you must first set up your environment by
+installing the necessary tools. This page explains the steps for setting up your
+development environment.
+
+1. [Install the gcloud CLI.](https://cloud.google.com/sdk/docs/install)
+1. In the Google Cloud console, on the project selector page, select or
+   [create a Google Cloud project](https://cloud.google.com/resource-manager/docs/creating-managing-projects).
+   {{< hint info >}}
+
+   **Note:** If you don't already have a project to use for testing changes to
+   the Terraform providers, create a project instead of selecting an existing
+   poject. After you finish these steps, you can delete the project, removing
+   all resources associated with the project.
+
+   {{< /hint >}}
+   {{< button href="https://console.cloud.google.com/projectselector2/home/dashboard" >}}Go to project selector{{< /button >}}
+1. Make sure that billing is enabled for your Google Cloud project. Learn how to
+   [check if billing is enabled on a project](https://cloud.google.com/billing/docs/how-to/verify-billing-enabled).
+
+   {{< hint warning >}}
+   If you are familiar with Docker or Podman, you may want to use the experimental
+   [container-based environment]({{< ref "/reference/make-commands.md#container-based-environment" >}})
+   instead of the following steps.
+   {{< /hint >}}
```

Comment:
This doesn't belong here... we could move it to the top of the page but I think we could also just remove the reference to it at this point; it's not really recommended.

```suggestion
```